app/models/course/assessment.rb

```diff
@@ -127,6 +127,14 @@ def password_protected?
     password.present?
   end
 
+  def submission_downloadable?
+    questions.where(
```

Comment:
Since you implemented `downloadable?` now, here better use it as well. Otherwise we need to update here again if the models are updated.

Performance is not a main concern here because this page is only for staff.